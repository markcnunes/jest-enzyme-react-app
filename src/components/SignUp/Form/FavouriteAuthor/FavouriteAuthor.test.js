import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import { act } from "react-dom/test-utils";

jest.mock("./_fetchAuthors", () => {
  return {
    __esModule: true,
    default: async () => [
      {
        id: 1,
        name: "Test Author 1"
      },
      {
        id: 2,
        name: "Test Author 2"
      }
    ]
  };
});

test("matches snapshot", async () => {
  const FavouriteAuthor = require("./FavouriteAuthor").default;

  let wrapper;

  await act(async () => {
    wrapper = mount(<FavouriteAuthor />);
  });

  wrapper.update();

  expect(toJSON(wrapper)).toMatchSnapshot();
});
