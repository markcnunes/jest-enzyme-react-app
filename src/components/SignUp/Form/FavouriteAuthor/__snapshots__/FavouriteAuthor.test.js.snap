// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ForwardRef>
  <styled.div>
    <div
      className="sc-AxiKw beNuzD"
    >
      <styled.select
        disabled={false}
        name="plan"
      >
        <select
          className="sc-AxirZ iDiCCD"
          disabled={false}
          name="plan"
        >
          <option
            value=""
          >
            (choose one)
          </option>
          <option
            key="1"
            value={1}
          >
            Test Author 1
          </option>
          <option
            key="2"
            value={2}
          >
            Test Author 2
          </option>
        </select>
      </styled.select>
      <styled.div>
        <div
          className="sc-AxjAm kDfPmt"
        />
      </styled.div>
    </div>
  </styled.div>
</ForwardRef>
`;
